src/equation_of_state.jl

```diff
@@ -39,3 +37,5 @@ end
 
 @inline buoyancy(i, j, k, grid, eos::LinearEquationOfState, grav, T, S) =
     @inbounds eos.βT * (T[i, j, k] - eos.T₀) - eos.βS * (S[i, j, k] - eos.S₀)
+
+NoEquationOfState() = LinearEquationOfState(βT=0, βS=0)
```

Comment:
Defining
```julia
@inline buoyancy(i, j, k, grid, eos::NoEquationOfState, grav, T, S) = nothing
```
would be more performant but the boost would be negligible.

This is nice otherwise.

Reply:
The problem right now is that the code uses the reference density by default. Therefore either we have to 

1. Include the reference density in "No equation of state", or
2. Finally change the code so that we use buoyancy perturbations rather than density perturbations, ala #236

Reply:
Sorry, I should also mention that `NoEquationOfState` is not a type --- but it should be, once we resolve the issue with reference densities. I don't like including the reference density when it is not used (the code should not be longer than it needs to be...). This also applies to the extra fields in `LinearEquationOfState` that are not used, like the reference pressure, or volumetric coefficient of thermal expansion for water.

Reply:
Ah fair enough. It is definitely terse which is nice.